q2_demux/tests/test_demux.py

```diff
@@ -776,31 +777,41 @@ def test_phred_score_out_of_range(self):
                 self.assertIn('<strong>Danger:</strong>', html)
 
     def test_inconsistent_sequence_length_single(self):
-        sequences = [('@s1/1 abc/1', 'GGGG', '+', 'YYYY'),
-                     ('@s2/1 abc/1', 'CCC', '+', 'PPP'),
-                     ('@s3/1 abc/1', 'AA', '+', 'PP'),
+        sequences = [('@s1/1 abc/1', 'GGGGGGG', '+', 'YYYYYYY'),
+                     ('@s2/1 abc/1', 'CCCCC', '+', 'PPPPP'),
+                     ('@s3/1 abc/1', 'AAA', '+', 'PPP'),
                      ('@s4/1 abc/1', 'T', '+', 'P')]
         bsi = BarcodeSequenceFastqIterator(self.barcodes, sequences)
 
         barcode_map = pd.Series(['AAAA', 'AACC'], index=['sample1', 'sample2'])
         barcode_map = qiime2.MetadataCategory(barcode_map)
 
         demux_data = emp_single(bsi, barcode_map)
-        with tempfile.TemporaryDirectory() as output_dir:
-            with self.assertRaisesRegex(ValueError,
-                                        'Observed sequences of length'):
+        lengths = [1, 3, 5, 7]
+        for n in range(1, 5):
```

Comment:
... also I kind of should have thought of this since I obviously thought about its implications ...

````PYTHON
_lengths = lengths[0:5-n] if n < 4 else [1]
````
🤦‍♂️